pages/docs/migrate-to-core-v6.en-US.mdx

```diff
@@ -0,0 +1,462 @@
+
+```typescript
+// Everything was in one giant esm-api package
+import { openmrsFetch, getPatient, getVisits } from "@openmrs/esm-api";
+```
+
+**The new way:**
+
+```typescript
+// Split into focused packages
+import { openmrsFetch } from "@openmrs/esm-api"; // Core HTTP/auth
+import { getPatient } from "@openmrs/esm-emr-api"; // Medical records
+```
+
+**Why we did this:** If your app only needs to make API calls but doesn't work with patients, you don't have to download the patient management code. Smaller bundles for everyone.
+
+#### Framework package cleanup
+
+We also modernized how packages expose their functionality:
+
+- **Before:** Deep imports like `@openmrs/esm-framework/src/internal/utils`
+- **After:** Clean public API through proper `exports` field
+- **Benefit:** Better TypeScript support, clearer boundaries, easier maintenance
+
+### Testing: Why your Jest config needs updates
+
+Core v6 embraces ES modules, which Jest historically struggled with. Here's how we solved it:
+
+#### The dual mock system
+
+The framework now ships with two sets of mocks:
+
+```bash
+@openmrs/esm-framework/
+├── mock.ts        # For ES modules (Vitest, modern Jest)
+├── mock-jest.ts   # For CommonJS (traditional Jest)
+└── package.json   # Automatically picks the right one
+```
+
+When you import `@openmrs/esm-framework/mock`, the framework automatically gives you the right version based on how you're importing it.
+
+#### Core framework uses Vitest now
+
+We migrated the framework itself from Jest to Vitest because:
+
+- **Native ES module support** - no complex configuration needed
+- **Faster test execution** - built on Vite's fast bundling
+- **Better developer experience** - clearer error messages
+
+Your apps can still use Jest! The Jest configuration updates in this guide ensure compatibility with our new ES module architecture.
+
+### Build system updates: Modern tooling
+
+#### Sass modernization
+
+- **New:** `sass-embedded` with modern compiler API
+- **Old:** Legacy node-sass (deprecated)
+- **Benefit:** Faster CSS compilation, better error messages
+
+#### Webpack evolution
+
+- **Module resolution:** Updated for ES modules
+- **CSS processing:** Improved pipeline for framework styles
+- **Development:** Better hot reloading and source maps
+
+### Migration strategy: Balancing change and stability
+
+#### Backward compatibility approach
+
+We designed the migration to be **incremental, not disruptive:**
+
+✅ **What stays the same:**
+
+- Your component APIs work unchanged
+- Configuration patterns are identical
+- Development workflow remains familiar
+
+⚠️ **What requires updates:**
+
+- Build configuration (Jest, Webpack)
+- Package versions and imports
+- TypeScript compilation setup
+
+#### Browser and environment support
+
+- **Modern browsers:** Full ES module support for optimal performance
+- **Older browsers:** Automatic transpilation maintains compatibility
+- **Node.js:** Requires Node 16+ for ES module support
```

Comment:
Few comments here:

* Although the framework _components_ are bundled as ESMs, we're still in the realm of using emulated ESMs in browser.
* I think we've set the minimum NodeJS version to 20.12+ (there is a work around, but you'd need to set a global Node flag)